surfacers/postgres/postgres.go

```diff
@@ -0,0 +1,295 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this postgres except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package postgres implements "postgres" surfacer. This surfacer type is in
+// experimental phase right now.
+package postgres
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/google/cloudprober/logger"
+	"github.com/google/cloudprober/metrics"
+
+	"database/sql"
+	"encoding/json"
+	configpb "github.com/google/cloudprober/surfacers/postgres/proto"
+	"github.com/lib/pq"
+	"strconv"
+	"time"
+)
+
+type label struct {
+	key   string
+	value string
+}
+
+type pgMetric struct {
+	time       time.Time
+	metricName string
+	value      string
+	labels     []label
+}
+
+// labelsJSON takes the label array and formats it for insertion into
+// postgres jsonb labels column, storing each label as k,v json object
+func (c pgMetric) labelsJSON() (string, error) {
+	m := make(map[string]string)
+	for _, l := range c.labels {
+		m[l.key] = l.value
+	}
+
+	bs, err := json.Marshal(m)
+	if err != nil {
+		return "", err
+	}
+
+	return string(bs), nil
+}
+
+func newPGMetric(t time.Time, metricName, val string, labels []label) pgMetric {
+	return pgMetric{
+		time:       t,
+		metricName: metricName,
+		value:      val,
+		labels:     labels,
+	}
+}
+
+type pgDistribution struct {
+	*metrics.DistributionData
+
+	metricName string
+	labels     []label
+	timestamp  time.Time
+}
+
+// sumMetric creates the correctly named metric
+// representing sum of the distribution
+func (d pgDistribution) sumMetric() pgMetric {
+	return newPGMetric(
+		d.timestamp,
+		d.metricName+"_sum",
+		strconv.FormatFloat(d.Sum, 'f', -1, 64),
+		d.labels,
+	)
+}
+
+// countMetric creates the correctly named metric representing
+// the count of the distribution
+func (d pgDistribution) countMetric() pgMetric {
+	return newPGMetric(
+		d.timestamp,
+		d.metricName+"_count",
+		strconv.FormatInt(d.Count, 10),
+		d.labels,
+	)
+}
+
```

Comment:
Can you inline these two functions? They are doing very little and are quire easy to follow, and are called just once.